test/serializers/associations_test.rb

```diff
@@ -290,6 +289,111 @@ def test_illegal_conditional_associations
           assert_match(/:if should be a Symbol, String or Proc/, exception.message)
         end
       end
+
+      class InheritedSerializerTest < ActiveSupport::TestCase
+        InheritedPostSerializer = Class.new(PostSerializer) do
+          belongs_to :author, custom_option: true
+          has_many :comments, custom_option: true, key: :reviews
+        end
+
+        InheritedAuthorSerializer = Class.new(AuthorSerializer) do
+          has_many :posts, custom_option: true
+          has_one :roles, custom_option: true, key: :new_roles
+          has_one :bio, custom_option: true
+        end
+
+        def setup
+          @author = Author.new(name: 'Steve K.')
+          @author.bio = Bio.new(id: 1, content: 'AMS Contributor')
+          @blog = Blog.new(name: 'AMS Blog')
+          @post = Post.new(title: 'New Post', body: 'Body')
+          @post.blog = @blog
+          @post.author = @author
+
+          @post_serializer = PostSerializer.new(@post, custom_options: true)
+          @author_serializer = AuthorSerializer.new(@author)
+          @inherited_post_serializer = InheritedPostSerializer.new(@post, custom_options: true)
+          @inherited_author_serializer = InheritedAuthorSerializer.new(@author)
+        end
+
+        def test_redefined_has_many_and_has_one
+          without_redefine = lambda do
+            associations = {}
+            @author_serializer.associations.each do |ass|
+              associations[ass.name] = ass.options[:custom_option]
+            end
+
+            expected_associations = {
+              posts: nil,
+              roles: nil,
+              bio: nil
+            }
+
+            assert_equal(associations, expected_associations)
+            assert_equal(@author_serializer.associations.count, 3)
+          end
+
+          without_redefine.call
```

Comment:
well, why not split apart the tests? having so much setup and assert in a single test is just asking to be split up.